### PR TITLE
[New] Create convex hull around geometries

### PR DESCRIPTION
**Pull request by CalebRas**
## Description

This PR implements `Create convex hull around geometries` in `Geometry` category.
URL to README:  https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-CreateConvexHullAroundGeometries/Shared/Samples/Create%20convex%20hull%20around%20geometries

## Linked Issue(s)

- `swift/issues/4206`

## How To Test

* Tap the "Create" button.
* Tap the "Union" toggle switch.
* Tap the "Reset" button.

## Screenshots

create-convex-hull-around-geometries

### CODE REVIEW

File: Shared/Samples/Create convex hull around geometries/CreateConvexHullAroundGeometriesView.swift [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                        }
+                    }
+                }
+            }
+    }
+}
+
+private extension CreateConvexHullAroundGeometriesView {
+    /// Creates convex hulls from the example polygons.
+    func createConvexHulls() {
```
Comment (yo1995) [inline Shared/Samples/Create convex hull around geometries/CreateConvexHullAroundGeometriesView.swift:70]: Better making this a pure function `func makeConvexHulls(for geometries: [Geometry], unioned: Bool) -> [Geometry]` so it is easier for users to copy and paste. And handle the graphic overlay add/delete outside the method.

File: Shared/Samples/Create convex hull around geometries/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+## Use case
+
+Creating a convex hull allows for analysis to define the polygon with the least possible perimeter that encloses a group of geometric shapes. As a visual analogy, consider a set of nails in a board where the convex hull is a rubber band stretched around the outermost nails.
+
+## How to use the sample
+
+Tap on the "Create" button to create convex hull(s) from the polygon graphics. If the 'Union' toggle is on, the resulting output will be one polygon being the convex hull for the two input polygons. If the 'Union' toggle is off, the resulting output will have two convex hull polygons - one for each of the two input polygons. Tap "Reset" to start over and remove the convex hull(s).
+
+
```
Comment (yo1995) [inline Shared/Samples/Create convex hull around geometries/README.md:15]:
```suggestion
```

File: Shared/Samples/Create convex hull around geometries/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+![Image of create convex hull around geometries](create-convex-hull-around-geometries.png)
+
+## Use case
+
+Creating a convex hull allows for analysis to define the polygon with the least possible perimeter that encloses a group of geometric shapes. As a visual analogy, consider a set of nails in a board where the convex hull is a rubber band stretched around the outermost nails.
+
+## How to use the sample
+
+Tap on the "Create" button to create convex hull(s) from the polygon graphics. If the 'Union' toggle is on, the resulting output will be one polygon being the convex hull for the two input polygons. If the 'Union' toggle is off, the resulting output will have two convex hull polygons - one for each of the two input polygons. Tap "Reset" to start over and remove the convex hull(s).
```
Comment (yo1995) [inline Shared/Samples/Create convex hull around geometries/README.md:13]:
```suggestion
Tap on the "Create" button to create convex hull(s) from the polygon graphics. If the "Union" toggle is on, the resulting output will be one polygon being the convex hull for the two input polygons. If the "Union" toggle is off, the resulting output will have two convex hull polygons - one for each of the two input polygons. Tap "Reset" to start over and remove the convex hull(s).
```

File: Shared/Samples/Create convex hull around geometries/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+## How it works
+
+1. Create an `Map` and display it in a `MapView`.
+2. Create two input polygon graphics and add them to a `GraphicsOverlay`.
+3. Call `GeometryEngine.convexHull(for:shouldMerge:)`, specifying a list of geometries for which to generate the convex hull. Set the `shouldMerge:` parameter to `true` to generate a convex hull for the union of the geometries. Set it to `false` to create a convex hull for each individual geometry.
+4. Loop through the returned geometries and add them to a Graphics Overlay to display on the map.
+
+## Relevant API
+
+* GeometryEngine.ConvexHull
```
Comment (yo1995) [inline Shared/Samples/Create convex hull around geometries/README.md:25]: Please use the qualified symbol name.

**Comment by yo1995** [comment]
It seems I cannot unassigned myself, so I'll assign Destiny instead. 🤐

File: Shared/Samples/Create convex hull around geometries/CreateConvexHullAroundGeometriesView.swift [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    
+    var body: some View {
+        MapView(map: map, graphicsOverlays: [convexHullGraphicsOverlay, geometriesGraphicsOverlay])
+            .toolbar {
+                ToolbarItemGroup(placement: .bottomBar) {
+                    Toggle("Union", isOn: $shouldUnion)
+                        .toggleStyle(.switch)
+                        .onChange(of: shouldUnion) { _ in
+                            createConvexHulls()
+                        }
```
Comment (des12437) [inline Shared/Samples/Create convex hull around geometries/CreateConvexHullAroundGeometriesView.swift]:
```suggestion
```
This toggle should only set the `shouldUnion` vale right? The convex hulls are already created when the `Create` button is tapped.

Comment (CalebRas) [reply Shared/Samples/Create convex hull around geometries/CreateConvexHullAroundGeometriesView.swift]: I implemented it with the idea that user could toggle the switch to see the results of unioning or not if there was already a convex hull was on the screen. Similar to the "Create buffers around points" sample, toggling shows the change without having to tap Reset and Create again. Doing so requires recreating the convex hull to show the change immediately. I updated this part a little bit implementing some of Ting's comments, but the functionality is the same. Let me know if you want the toggle to just update the shouldUnion instead.

Comment (des12437) [reply Shared/Samples/Create convex hull around geometries/CreateConvexHullAroundGeometriesView.swift]: The updated implementation looks good :+1:

**Review by des12437 — APPROVED** [review]

**Review by dfeinzimer — APPROVED** [review]

**Comment by yo1995** [comment]
Thanks Nimesh, I'll log an issue to improve.

**Pull request merged**